src/TraceEvent/ETWReloggerTraceEventSource.cs

```diff
@@ -407,7 +407,7 @@ private unsafe void Initialize(TraceEventNativeMethods.EVENT_RECORD* rawData)
                         // first event (real time case).   This is really a problem, as we need that information, but we will go ahead and
                         // try to initialize as best we can. 
 
-                        m_source.pointerSize = (RuntimeInformation.OSArchitecture == Architecture.X64 || RuntimeInformation.OSArchitecture == Architecture.Arm64) ? 8 : 4;
+                        m_source.pointerSize = IntPtr.Size;
```

Comment:
❔ What is the result you want when PerfView is run as a 32-bit process on a 64-bit operating system? The approach taken here will produce `4` in this case.

Reply:
:bulb: My recommendation is to use this:

```csharp
#if !NETSTANDARD1_6
m_source.pointerSize = Environment.Is64BitOperatingSystem;
#else
m_source.pointerSize = (RuntimeInformation.OSArchitecture == Architecture.X64 || RuntimeInformation.OSArchitecture == Architecture.Arm64) ? 8 : 4;
#endif
```

Reply:
Yes, I came to a similar conclusion.

Reply:
It sounds like it will make sense to bring back a **netstandard2.0** build, since that one could use the working API and most users will be able to leverage it. The maintenance cost would be almost nothing since the API surface area is already covered by other targets we are building.